### PR TITLE
bpo-40275: Use new test.support helper submodules in tests

### PR DESCRIPTION
**Pull request by shihai1991**
Use test helpers in the following test cases:
* tests of distutils
* test_buffer
* test_compile
* test_filecmp
* test_fileinput
* test_readline
* test_smtpnet
* test_structmembers
* test_tools

https://bugs.python.org/issue40275

### CODE REVIEW

File: Lib/test/test_fileinput.py [hunk 1]
```diff
@@ -39,7 +41,7 @@ class BaseTests:
     # temp file's name.
     def writeTmp(self, content, *, mode='w'):  # opening in text mode is the default
         fd, name = tempfile.mkstemp()
-        self.addCleanup(support.unlink, name)
+        self.addCleanup(safe_unlink, name)
```
Comment (vstinner) [inline Lib/test/test_fileinput.py]: I would prefer to use os_helper.unlink here, rather than changing the function name.

Comment (shihai1991) [reply Lib/test/test_fileinput.py]: done.
I use safe_unlink in here's  reason is that test_fileinput have import unlink as safe_unlink. 
if I import os_helper again, the imports looks like more duplicated~

Comment (vstinner) [reply Lib/test/test_fileinput.py]: It's common to import a module and import some symbols of the same module. "Inconsistencies" are common, especially in tests which are old :-)

Comment (vstinner) [reply Lib/test/test_fileinput.py]: For such PR which basically updates tests to point to new submodules, I prefer to leave the tests unchanged, don't change the coding style in the same PR.

Comment (vstinner) [reply Lib/test/test_fileinput.py]: (Also, PR which only changes coding style are usually rejected. In short, the coding style cannot be updated in Python code base :-))

Comment (shihai1991) [reply Lib/test/test_fileinput.py]: > (Also, PR which only changes coding style are usually rejected. In short, the coding style cannot be updated in Python code base :-))

Got it, Thanks. I have know the pain of code churn( not easy for finding the historical  record ;-) )

**Pull request merged**